gm4_mob_curing/data/gm4_mob_curing/functions/main.mcfunction

```diff
@@ -0,0 +1,9 @@
+#execute on red mushrooms to check for cows around them
+execute at @e[type=item,nbt={Item:{id:"minecraft:red_mushroom"}},limit=1,sort=random] if entity @e[type=item,distance=..1.5,limit=1,nbt={Item:{id:"minecraft:brown_mushroom"}}] as @e[type=cow,distance=..1.5,limit=1,nbt={ActiveEffects:[{Id:2b}]}] unless data entity @s NoAI at @s run function gm4_mob_curing:cure/mooshroom
+execute at @e[type=item,nbt={Item:{id:"minecraft:golden_carrot"}},limit=1,sort=random] as @e[type=zombified_piglin,distance=..1.5,limit=1] unless data entity @s NoAI at @s run function gm4_mob_curing:cure/zombified_check
+execute at @e[type=item,nbt={Item:{id:"minecraft:golden_carrot"}},limit=1,sort=random] as @e[type=zoglin,distance=..1.5,limit=1,nbt={ActiveEffects:[{Id:1b}]}] unless data entity @s NoAI at @s run function gm4_mob_curing:cure/zoglin
+execute at @e[type=item,nbt={Item:{id:"minecraft:golden_apple"}},limit=1,sort=random] as @e[type=witch,distance=..1.5,limit=1,nbt={ActiveEffects:[{Id:18b}]}] unless data entity @s NoAI at @s run function gm4_mob_curing:cure/witch
```

Comment:
You're searching through all items entities three times here. A single `execute at @e[type=item...] run function...` in combination with `@s` checks inside said function would definitely be beneficial. This would also remove the double lookup for `@e[type=item,nbt={Item:{id:"minecraft:golden_carrot"}},limit=1,sort=random]`.

Furthermore, `limit=1` makes it so that only one cow can be converted per clock pulse (globally!); This seems unintentional.

Reply:
The cow limiter is only per `@e[item`, no?

Reply:
This is `main.mcfunction`. Each of these lines will only act as a single item per clock pulse.